spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -764,6 +776,7 @@ protected void doStart() throws Exception {
 						+ Arrays.asList(queueNames));
 			}
 		}
+		checkMismatchedQueues();
```

Comment:
Actually, the code in `onInit()` is a bit dubious.

First it randomly chooses the first admin if there are more than one.

Second, if there are none, it creates one, sets the AC but doesn't call `afterPropertieSet()` - which is likely a good thing since we shouldn't declare all the queues etc in that case.

I will rework that code in another commit.